manifests/init.pp

```diff
@@ -1,25 +1,46 @@
-# Default strongSwan class.
+# @summary Installs and manages strongSwan
+#
+# @param package_name
+#   The strongswan package name.
+# @param package_ensure
+#    The desired package state. Can be used to install `latest` or a specific version etc.
+# @param service_name
+#    The strongswan service name.
+# @param service_ensure
+#    The desired `ensure` state for the service.
+# @param service_enable
+#    The desited `enable` state for the service.
+# @param ipsec_d_dir
+#    The directory for the ipsec configuration files.  The default is operating system specific and you should not need to override this setting.
+# @param ipsec_conf
+#    The location of the `ipsec.conf` file. The default is operating system specific and you should not need to override this setting.
+# @param ipsec_secrets
+#    The location of the `ipsec.secrets` file. The default is operating system specific and you should not need to override this setting.
+# @param charon_conf
```

Comment:
I think I'll open a separate PR to remove this parameter after I've thought a bit longer about how to manage the main charon.conf file.